subiquity/common/serialize.py

```diff
@@ -194,6 +194,10 @@ def _deserialize_attr(self, annotation, value, metadata, path):
             args = {}
             fields = {field.name: field for field in attr.fields(annotation)}
             for key in value.keys():
+                if key not in fields and key == '$type':
+                    # Union types can contain a '$type' field that is not
+                    # actually one of the keys.
+                    continue
```

Comment:
Ugh is this a generic problem with List[Union]? :( Sounds like that can be fixed in a separate branch (with tests <wink>)

Reply:
https://github.com/canonical/subiquity/pull/1177